### PR TITLE
[NIFI-9381] Test Does not pass due to NonDex

### PR DESCRIPTION
**Pull request by SpiceOwlf**
#### Description of PR

_fixes bug NIFI-9381._ 
Tests related to _initCompositeUserGroupProvider_ under _CompositeUserGroupProviderTestBase_ may express non-deterministic behavior due to the nature of HashMap. The fix is changing HashMap with LinkedHashMap to ensure deterministic behavior.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-authorization-providers/src/test/java/org/apache/nifi/authorization/CompositeUserGroupProviderTestBase.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 import org.junit.Assert;
 
 import java.util.HashMap;
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-authorization-providers/src/test/java/org/apache/nifi/authorization/CompositeUserGroupProviderTestBase.java:24]: According to the automated build, this import is no longer used and should be removed.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for the contribution @SpiceOwlf! This looks like a straightforward improvement to ensure deterministic test behavior. Please note the automated build checkstyle warning due to an unused import. If you can remove the unused import, that should resolve the problem.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-authorization-providers/src/test/java/org/apache/nifi/authorization/CompositeUserGroupProviderTestBase.java (hunk 1 above)
Comment (SpiceOwlf) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-authorization-providers/src/test/java/org/apache/nifi/authorization/CompositeUserGroupProviderTestBase.java]: Absolutely, I just removed HashMap. Hope the workflow will pass this time!

**Comment by exceptionfactory** [comment]
Thanks for updating the import @SpiceOwlf. After closer inspection, it is not clear how changing to use the LinkedHashMap resolves the test failure in the Jira issue.  The failing assertion compares the number of users associated with the group, and that does not appear directly related to the mocked properties provided through the Map.  Each property set in the Map should have a unique number, and order should not matter.  Can you describe how using LinkedHashMap resolves the potential non-deterministic test behavior?

**Comment by SpiceOwlf** [comment]
Here are what I observed:
While using HashMap, NonDex will complain that `java.lang.AssertionError: expected:<1> but was:<2>`, and the error was cast from `CompositeUserGroupProviderTest.testMultipleProvidersWithConflictingUsers:135->CompositeUserGroupProviderTestBase.testUserGroupProviderTwo:122`

With a little bit more digging, `userGroupProvider.getGroup(GROUP_2_IDENTIFIER).getUsers().size()` is the cause of this error, and  `userGroupProvider.getGroup(GROUP_2_IDENTIFIER).getUsers()` will return `<[user-identifier-4, user-identifier-1]>` or `<[user-identifier-3]>` depending on the random seed NonDex received. Changing HashMap to LinkedHashMap eliminate such random behavior. It ensures that `initCompositeUserGroupProvider` has deterministic order while using `mockProperties`.  Please let me know if my explaination makes sense or not. I will also attach the steps to reproduce this error later.

**Review by exceptionfactory — APPROVED** [review]
Thanks for the additional background @SpiceOwlf, that is helpful. +1 Merging.

**Pull request closed without merging**